src/Microsoft.SqlTools.ServiceLayer/Scripting/Contracts/ScriptingOptions.cs

```diff
@@ -0,0 +1,178 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting.Contracts
+{
+    /// <summary>
+    /// Defines the scripting options.
+    /// </summary>
+    public class ScriptOptions
+    {
+        /// <summary>
+        /// Generate ANSI padding statements
+        /// </summary>
+        public string ScriptAnsiPadding { get; set; }
+
+        /// <summary>
+        /// Append the generated script to a file
+        /// </summary>
+        public string AppendToFile { get; set; }
+
+        /// <summary>
+        /// Check that an object with the given name exists before dropping or altering or that an object with the given name does not exist before creating.
+        /// </summary>
+        public string CheckForObjectExistence { get; set; }
+
+        /// <summary>
+        /// Continue to script if an error occurs. Otherwise, stop.
+        /// </summary>
+        public string ContinueScriptingOnError { get; set; }
+
+        /// <summary>
+        /// Convert user-defined data types to base types.
+        /// </summary>
+        public string ConvertUDDTToBaseType { get; set; }
```

Comment:
I'm using the same names used by SMO. I'm my mind, this makes it easier to reason over all of options. I'm going to leave as-is for now unless folks feel strongly.

---
In reply to: [112277263](https://github.com/Microsoft/sqltoolsservice/pull/316#discussion_r112277263) [](ancestors = 112277263)